util/gitutil/git_cli.go

```diff
@@ -216,7 +215,7 @@ func (cli *GitCLI) Run(ctx context.Context, args ...string) (_ []byte, err error
 					continue
 				}
 			}
-			return buf.Bytes(), errors.Errorf("git error: %s\nstderr:\n%s", err, errbuf.String())
+			return buf.Bytes(), fmt.Errorf("git error: %w\nstderr:\n%s", err, errbuf.String())
```

Comment:
This will fail in the linter because `fmt.Errorf`

I think we should also detect cancellation in here as I don't think `err` is `context.Canceled` but if the function errored because `ctx` was canceled that should be the root error. If this happens we could set it to `context.Canceled` with extra wrapping on any error or I'm not sure if it makes sense to type check original error first.

Reply:
> This will fail in the linter because fmt.Errorf

Why would that fail the linter?

All these does is preserve the root error if it comes through.

Reply:
We have a feature that `--debug` on errored build shows stacktraces for the error across all components (and saves in daemon logs if debug logs enabled). Lint rule is to make sure we don't accidentally use `fmt.Errorf` and create an error that doesn't have a stacktrace.